estimagic/dashboard/convergence_plot.py

```diff
@@ -0,0 +1,157 @@
+    plots = [fitness_plot] + param_plots
+    datasets = [fitness_data] + param_data
+    return plots, datasets
+
+
+def update_convergence_plot(doc, queue, datasets):
+    """
+    Check for new param values and update the plot.
+
+    This function is called in a never ending while loop in _update_dashboard.
+
+    Args:
+        doc (bokeh Document):
+            document instance where the Dashboard will be stored.
+            Note this must stay the first argument for the bokeh FunctionHandler
+            to work properly.
+
+        queue (Queue):
+            queue to which originally the parameters DataFrame is supplied and to which
+            the updated parameter Series will be supplied later.
+
+        datasets (list):
+            list of ColumnDataSource storing earlier parameter iterations.
+
+    """
+    fitness_data, param_data = datasets
+
+    if queue.qsize() > 0:
+        new_params, fitness = queue.get()
+
+        doc.add_next_tick_callback(
+            partial(_update_convergence_plot, data=param_data, new_values=new_params)
+        )
+        doc.add_next_tick_callback(
+            partial(_update_convergence_plot, data=fitness_data, new_values=fitness)
+        )
+
+
+def _fitness_plot(initial_fitness):
+    fitness_data = ColumnDataSource(
+        data=_convert_sr_for_cds(sr=initial_fitness, iteration=0)
+    )
+    fitness_p = _wide_figure(title="Fitness")
+    fitness_p.line(
+        source=fitness_data,
+        x="iteration",
+        y="fitness",
+        line_width=1,
+        name="fitness",
+        color="firebrick",
+    )
+
+    return fitness_p, fitness_data
+
+
+def _parameter_plots(param_df):
+    """
+    Create the plots that show the convergence of (groups of) parameters.
+
+    ToDo: split up convergence plot depending on column in param_df.
+    ToDo: plot upper and lower bounds.
+    ToDo: only plot parameters that are not fixed.
+
+    Args:
+        param_df (pandas DataFrame):
+            DataFrame with the initial parameter values, constraints etc.
+
+
+    """
+
+    assert "iteration" not in param_df.index, (
+        "Estimagic uses the key 'iteration'. "
+        + "Therefore, it may not be used in the index of the parameter DataFrame."
+    )
+
+    colors = _choose_color_palettes(param_df)
+
+    first_entry = _convert_sr_for_cds(sr=param_df["value"], iteration=0)
+    param_data = ColumnDataSource(data=first_entry)
+
+    conv_p = _wide_figure(title="Parameter Values")
+
+    _add_convergence_lines(figure=conv_p, param_data=param_data, colors=colors)
+
+    return [conv_p], [param_data]
+
+
+def _choose_color_palettes(param_df):
+    # color tone palettes: bokeh.palettes.Blues9, Greens9, Reds9, Purples9.
+    long_colors = bokeh.palettes.Viridis256
+    return random.sample(long_colors, len(param_df))
+
+
+def _add_convergence_lines(figure, param_data, colors):
+    line_names = [str(x) for x in sorted(param_data.column_names) if x != "iteration"]
```

Comment:
What are the parameter names if we have a MultiIndex?

Reply:
They are tuples. So if for one entry the first level of the Index was 'a' and for the second it was 1, then the parameter name (column name in the ColumnDataSource is ('a', 1).

Reply:
That's ok for now and we can also use it in the legend.